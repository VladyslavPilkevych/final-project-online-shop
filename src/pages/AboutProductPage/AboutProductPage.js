/* eslint-disable no-underscore-dangle */
import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import { useMatch, useLocation, useNavigate } from 'react-router-dom';
import AboutProductDetailsTabText from '../../components/AboutProductTabs/AboutProductDetailsTabText/AboutProductDetailsTabText';
import AboutProductCommonTabText from '../../components/AboutProductTabs/AboutProductCommonTabText/AboutProductCommonTabText';
import AboutProductImage from '../../components/AboutProductTabs/AboutProductImage/AboutProductImage';

import Button from '../../components/Button/Button';

import { getProduct } from '../../store/actionCreators/productsAC';
import styles from './AboutProductPage.module.scss';

function AboutProductPage() {
  const [activeTab, setActiveTab] = useState('tab1');
  // const navigate = useNavigate();

  const handleTab1 = () => {
    setActiveTab('tab1');
  };

  const handleTab2 = () => {
    setActiveTab('tab2');
  };
  const product = useSelector((state) => state.products.activeProduct);
  const location = useLocation();
  const newLocation = location.pathname.split('/').slice(-1);

  console.log(newLocation);

  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getProduct(...newLocation));
  }, []);

  return (
    <>
      {/* <div>
        AboutProductPage
        {product.imageUrls}
        <p>{product.name}</p>
        <p>{product.model}</p>
        <p>{product.quantity}</p>
        <p>{product.storage}</p>
        <p>{product.color}</p>
        <p>{product.display}</p>
        <p>{product.currentPrice}</p>
        <p>{product.previousPrice}</p>
      </div> */}
      {/* <div className={styles.goBackBtn}>
  <Button className={styles.aboutProductCartBtn} handleClick={() => navigate(-1)}>Go Back</Button>
      </div> */}
      <hr className={styles.line} />
      <div className={`${styles.aboutProductTabMenuWrapper} ${styles.aboutProductWrapper}`}>
        <ul className={styles.aboutProductTabMenu}>
          <li
            className={activeTab === 'tab1' ? `${styles.active}` : ''}
            role="menuitem"
            onClick={handleTab1}
          >
            About Product
          </li>
          <li
            className={activeTab === 'tab2' ? `${styles.active}` : ''}
            role="menuitem"
            onClick={handleTab2}
          >
            Details
          </li>
        </ul>
        <div className={styles.aboutProductCartBtnWrapper}>
          <button type="button" className={styles.aboutProductCartBtn} onClick={() => console.log('Add to Cart', product._id)}>Add to Cart</button>
        </div>
      </div>
      <hr className={styles.line} />
      <div className={`${styles.aboutProducTextAndImageWrapper} ${styles.aboutProductWrapper}`}>
        <div className={styles.aboutProductTextWrapper}>
          {activeTab === 'tab1' ? (
            <AboutProductCommonTabText product={product} />
          ) : (
            <AboutProductDetailsTabText product={product} />
          )}
        </div>
        <div className={styles.aboutProducImageWrapper}>
          <AboutProductImage urls={product.imageUrls} model={product.model} id={product._id} />
        </div>
      </div>
    </>

  );
}

export default AboutProductPage;
