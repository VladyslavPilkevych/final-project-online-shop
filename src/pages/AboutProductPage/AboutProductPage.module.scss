@import '../../assets/styles/global.module.scss';

.aboutProductWrapper {
  max-width: 1398px;
  padding: 20px;
  margin: 0 auto;
}

.line {
  size: 0.5px;
  color: $dark-grey-text;
  width: 100%;
}
.goBackBtn {
  margin: 15px;
}
.aboutProductTabMenuWrapper {
  display: flex;
  justify-content: space-between;
}

.costWrapper {
  font-weight: bold;
  margin-right: 20px;
}

@media screen and (max-width: 768px) {
  .aboutProductTabMenuWrapper {
    padding: 0;
  }
}

.aboutProductTabMenu {
  color: black;
  font-family: $fonts;
  font-style: normal;
  font-weight: 600;
  font-size: 14px;
  line-height: 21px;
  margin-top: 41px;
  margin-bottom: 38px;
  display: flex;
}

.priceWrapper {
  display: flex;
}

.priceTitle {
  font-family: $fonts;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 21px;
  margin-right: 15px;
}
.priceAmount {
  font-family: $fonts;
  font-style: bold;
  font-weight: 600;
  font-size: 14px;
  line-height: 21px;
  margin-right: 20px;
}

.aboutProductCartBtnWrapper {
  display: flex;
  align-items: center;
}

.aboutProductCartBtn {
  // margin-top: 30px;
  background-color: #2e58ff;
  color: $white;
  border-radius: 50px;
  padding: 12px 30px;
  position: relative;
  display: inline-block;
  transition: 0.5s;
  border: none;
}
.aboutProductCartBtn:hover {
  background-color: darken(#1d4aff, 5%);
  box-shadow: 2px 2px 15px 5px #7b6aff;
  color: $white;
  cursor: pointer;
}

.aboutProducTextAndImageWrapper {
  display: flex;
  justify-content: space-around;
  max-width: 95%;
}

@media screen and (max-width: 768px) {
  .aboutProducTextAndImageWrapper {
    display: block;
    text-align: center;
  }
}

.aboutProductTextWrapper {
  background-color: $bg-white-color;
  width: 60%;
  padding: 37px 60px 25px 60px;
  order: -1;
}

@media screen and (max-width: 768px) {
  .aboutProductTextWrapper {
    order: 2;
    width: inherit;
    padding: 10px 20px 10px 20px;
  }
}

.aboutProducImageWrapper {
  width: 35%;
  display: flex;
  justify-content: center;
}

@media screen and (max-width: 768px) {
  .aboutProducImageWrapper {
    width: 100%;
  }
}

.aboutProductTabMenuWrapper ul li {
  cursor: pointer;
  // margin-right: 30px;
  font-family: $fonts;
  font-style: normal;
  font-weight: 600;
  font-size: 14px;
  line-height: 21px;
  color: $grey;
  margin-right: 20px;
}
.aboutProductTabMenuWrapper ul li.active {
  color: $black;
  border-bottom: 2px solid $blue;
}

.addToCartIconWrapper {
  cursor: pointer;
  margin: 150px 10px;
}
