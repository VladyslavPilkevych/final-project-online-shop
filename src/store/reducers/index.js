import { combineReducers } from 'redux';
import preloaderReducer from './preloaderReducer';
import productsReducer from './productsReducer';
import menuReducer from './menuReducer';
import userReducer from './userReducer';
import searchReducer from './searchReducer';
import cartReducer from './cartReducer';
import miniMenuReducer from './miniMenuReducer';
import filtersCategoriesReducer from './filtersCategoriesReducer';

const reducer = combineReducers({
  loader: preloaderReducer,
  products: productsReducer,
  menu: menuReducer,
  miniMenu: miniMenuReducer,
  filtersCategories: filtersCategoriesReducer,
  user: userReducer,
  search: searchReducer,
  cart: cartReducer,
});

export default reducer;
