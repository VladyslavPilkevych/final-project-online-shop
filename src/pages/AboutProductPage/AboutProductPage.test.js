import { render } from '@testing-library/react';
import AboutProductPage from './AboutProductPage';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../store';

function Component() {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <AboutProductPage />
      </Provider>
    </BrowserRouter>
  );
}

describe('Snapshot AboutProductPage should render', () => {
  test('Snapshot AboutProductPage is rendered', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});