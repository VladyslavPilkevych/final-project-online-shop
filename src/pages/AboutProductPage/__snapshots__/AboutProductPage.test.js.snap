// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot AboutProductPage should render Snapshot AboutProductPage is rendered 1`] = `
<DocumentFragment>
  <div
    class="aboutProductWrapper"
  >
    <div
      class="aboutProductTabMenuWrapper aboutProductWrapper"
    >
      <ul
        class="aboutProductTabMenu"
      >
        <li
          class="active"
          role="menuitem"
        >
          About Product
        </li>
        <li
          class=""
          role="menuitem"
        >
          Details
        </li>
      </ul>
      <div
        class="aboutProductCartBtnWrapper"
      >
        <p
          class="costWrapper"
        >
          Cost: $NaN
        </p>
        <button
          class="aboutProductCartBtn [object Object]"
          type="button"
        >
           Add to Cart
        </button>
      </div>
    </div>
    <hr
      class="line"
    />
    <div
      class="aboutProducTextAndImageWrapper aboutProductWrapper"
    >
      <div
        class="aboutProducImageWrapper"
      >
        <div
          class="imageSideWrapper"
        >
          <div
            class="imageWrapper"
          >
            <div
              class="slick-slider slick-initialized"
            >
              <div
                class="slick-list"
              >
                <div
                  class="slick-track"
                  style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="aboutProductTextWrapper"
      >
        <div>
          <h1
            class="productTitle"
          >
             
          </h1>
          <p
            class="productDescription"
          />
          <div
            class="colorsWrapper"
          />
          <p
            class="itemNo"
          >
            Article:
          </p>
          <p
            class="contactUsText"
          >
            Have a Question? 
            <a
              href="/contact-us"
            >
              Contact us
            </a>
          </p>
        </div>
      </div>
    </div>
    <div>
      <div
        class="advantagesWrapper"
      >
        <div
          class="advantagesCard"
        >
          <img
            alt="advantages-icon"
            class="advantagesIcon"
            src="account-icon.png"
          />
          <h3
            class="advantagesTitle"
          >
            Product Support
          </h3>
          <p
            class="advantagesText"
          >
            Up to 3 years on-site warranty available for your peace of mind.
          </p>
        </div>
        <div
          class="advantagesCard"
        >
          <img
            alt="advantages-icon"
            class="advantagesIcon"
            src="support-icon.png"
          />
          <h3
            class="advantagesTitle"
          >
            Personal Account
          </h3>
          <p
            class="advantagesText"
          >
            With big discounts, free delivery and a dedicated support specialist.
          </p>
        </div>
        <div
          class="advantagesCard"
        >
          <img
            alt="advantages-icon"
            class="advantagesIcon"
            src="saving-icon.png"
          />
          <h3
            class="advantagesTitle"
          >
            Amazing Savings
          </h3>
          <p
            class="advantagesText"
          >
            Up to 70% off new Products, you can be sure of the best price.
          </p>
        </div>
      </div>
    </div>
    <div
      class="Toastify"
    />
  </div>
</DocumentFragment>
`;
